src/common_swagger_api/schema/metadata/tags.clj

```diff
@@ -0,0 +1,101 @@
+(def DeleteUserTagsSummary "Delete Tags Defined by a User")
+(def DeleteUserTagsDescription "This endpoint deletes all tags defined by the current user. Corresponding attached tags will also be deleted.")
+(def PostTagSummary "Create a Tag")
+(def PostTagDescription "This endpoint creates a tag for use by the authenticated user.")
+(def DeleteTagSummary "Delete a Tag")
+(def DeleteTagDescription "This endpoint allows a user tag to be deleted, detaching it from all metadata.")
+(def PatchTagSummary "Update Tag Labels/Descriptions")
+(def PatchTagDescription "This endpoint allows a tag's label and description to be modified by the owning user.")
+
+(s/defschema TagTypeEnum {:type (describe (s/enum "attach" "detach") "Whether to attach or detach the provided set of tags to the file/folder")})
+
+(s/defschema UpdateAttachedTagsQueryParams
+  (merge StandardDataItemQueryParams
+         TagTypeEnum))
+
+(s/defschema TagSuggestQueryParams
+  (merge StandardUserQueryParams
+         {:contains
+          (describe String "The value fragment")
+
+          (s/optional-key :limit)
+          (describe TagSuggestLimit
+                    "The maximum number of suggestions to return. No limit means return all")}))
+
+(s/defschema TagId {:id TagIdPathParam})
+
+(s/defschema Tag
+  {:id
+   (describe UUID "The service-provided UUID associated with the tag")
+
+   :value
+   (describe TagValueString "The value used to identify the tag, at most 255 characters in length")
+
+   (s/optional-key :description)
+   (describe String "The description of the purpose of the tag")})
+
+(s/defschema TagRequest
+  (describe (dissoc Tag :id) "The user tag to create."))
+
+(s/defschema TagUpdateRequest
+  (describe (->optional-param TagRequest :value) "The tag fields to update."))
+
+(s/defschema TagList
+  {:tags (describe [Tag] "A list of Tags")})
+
+(s/defschema TagIdList
+  {:tags (describe [UUID] "A list of Tag UUIDs")})
+
+(s/defschema AttachedTagTarget
+  {:id   (describe UUID "The target's UUID")
+   :type (describe DataTypeEnum "The target's data type")})
+
+(s/defschema TagDetails
+  (merge Tag
+         {:owner_id    (describe String "The owner of the tag")
+          :public      (describe Boolean "Whether the tag is publicly accessible")
+          :created_on  (describe Long "The date the tag was created in ms since the POSIX epoch")
+          :modified_on (describe Long "The date the tag was last modified in ms since the POSIX epoch")}))
+
+(s/defschema AttachedTagDetails
+  (merge TagDetails
+         {:targets (describe [AttachedTagTarget] "A list of targets attached to the tag")}))
+
+(s/defschema AttachedTagsListing
+  {:tags (describe [AttachedTagDetails] "A list of tags and their attached targets")})
+
+(s/defschema ErrorResponseBadTagRequest
+  (assoc ErrorResponse
+    :error_code (describe (s/enum ERR_ILLEGAL_ARGUMENT ERR_NOT_UNIQUE) "Bad Tag Request error codes")))
```

Comment:
Currently this is only used in the `metadata` service, however it would be nice to also add `:responses` in the `terrain` endpoint as well.

Reply:
As in moving all the `:responses` bits from `metadata.routes.tags` to `common-swagger-api`?  If I do that, could I move over the entire `{200 ... 500 ... :default...}` responses map into `commons-swagger-api` or would I need to just move pieces of that map like the `:description` bits?

Reply:
Yes, you could move the entire `:responses` map.
For example:
https://github.com/cyverse-de/common-swagger-api/blob/bfc5ee4609dbce4d1e8aa57b176bd972cd20f052/src/common_swagger_api/schema/tools.clj#L322-L361
or
https://github.com/cyverse-de/common-swagger-api/blob/bfc5ee4609dbce4d1e8aa57b176bd972cd20f052/src/common_swagger_api/schema/data/tickets.clj#L86-L128